thriftbp/client_middlewares.go

```diff
@@ -313,3 +318,75 @@ var (
 	_ thrift.ClientMiddleware = SetDeadlineBudget
 	_ thrift.ClientMiddleware = BaseplateErrorWrapper
 )
+
+// PrometheusClientMiddleware returns middleware to track Prometheus metrics
+// specific to the Thrift client.
+//
+// It emits the following prometheus metrics:
+//
+// * thrift_client_active_requests gauge with labels:
+//
+//   - thrift_service: the serviceSlug arg
+//   - thrift_method: the method of the endpoint called
+//   - thrift_slug: the service being contacted
+//
+// * thrift_client_latency_seconds histogram with labels above plus:
+//
+//   - thrift_success: "true" if err == nil, "false" otherwise
+//
+// * thrift_client_requests_total counter with all labels above plus:
+//
+//   - thrift_exception_type: the human-readable exception type, e.g.
+//     baseplate.Error, etc
+//   - thrift_baseplate_status: the numeric status code from a baseplate.Error
+//     as a string if present (e.g. 404), or the empty string
+//   - thrift_baseplate_status_code: the human-readable status code, e.g.
+//     NOT_FOUND, or the empty string
+func PrometheusClientMiddleware(serviceSlug, serverSlug string) thrift.ClientMiddleware {
```

Comment:
`localServiceSlug, remoteServiceSlug`? (it took me awhile to work out that I think this is what they mean, so making the names match might help others)

Also "service" vs "server" is a bit confusing, so unless you have an overriding reason I would use "Service" for both